src/js/entities-service/entities/actions.js

```diff
@@ -104,10 +104,15 @@ const _Model = BaseModel.extend({
   },
   canEdit() {
     const currentUser = Radio.request('bootstrap', 'currentUser');
+    const owner = this.getOwner();
+    const currentUsersTeam = currentUser.getTeam();
+    const ownersTeam = owner.type === 'teams' ? owner : this.getOwner().getTeam();
```

Comment:
owner is always expected.  I think these `const`s should move to above the last `return` they're needed on because this is a very hot function on a list.

Reply:
actually I think the additional logic and look up should be inside:
```js
if (currentUser.can('work:team:manage') {
   const currentUsersTeam = currentUser.getTeam();
   const ownersTeam = owner.type === 'teams' ? owner : this.getOwner().getTeam();
}
```

we don't need to even attempt doing extra work until we know if we have to.

Reply:
That's been updated.